### PR TITLE
Add rule to capitalize acronyms

### PR DESCRIPTION
**Pull request by calda**
This PR adds a new rule that automatically capitalizes acronyms when the first character is capitalized (so acronyms are either consistently uppercase, or consistently lowercase).

```diff
- let destinationUrl: URL
- let urlRouter: UrlRouter
- let screenId: String
- let entityUuid: UUID

+ let destinationURL: URL
+ let urlRouter: URLRouter
+ let screenID: String
+ let entityUUID: UUID
```

I considered adding an option that reversed this transform as well (e.g. converted `URL` to `Url`), but this would conflict with `Foundation` types that obviously cannot be renamed.

### CODE REVIEW

File: Sources/Rules.swift [hunk 1]
```diff
@@ -5507,4 +5507,33 @@ public struct _FormatRules {
...
+        disabledByDefault: true,
+        options: ["acronyms"]
+    ) { formatter in
+        formatter.forEachToken { index, token in
+            for acronym in formatter.options.acronyms {
+                let find = acronym.capitalized
+                let replace = acronym.uppercased()
+
+                if token.string.contains(find) {
+                    let updatedText = token.string.replacingOccurrences(of: find, with: replace)
```
Comment (calda) [inline Sources/Rules.swift:5522]: aha this find/replace is too simplistic, because it will errantly convert `Identifier` to `IDentifiter` 😆

Comment (calda) [reply Sources/Rules.swift]: Made improvements to this in 92ac269

**Comment by nicklockwood** [comment]
Very cool, thanks!

**Comment by nicklockwood** [comment]
@calda do you think it would make sense to add an option to disable this for public properties? It seems dangerous to change those by default.

Also I guess this should exclude Codable, RawRepresentable enum cases, etc by default?

**Comment by calda** [comment]
Good question -- for `public` properties my thinking was that it would be safe to use this rule as long as you applied it to all code in your project. You're right that this seems problematic for frameworks who have downstream consumers. It does seem reasonable to have an option for this.

Great point about `Codable` and `RawRepresentable` types, I hadn't considered that -- those are definitely not safe to change automatically like this.

---

I tried enabling this rule for our codebase at Airbnb and I was swiftly defeated (we have too old Objective-C code that I would have had to update manually, and too much Swift code generated from backend models like GraphQL objects that can't be changed at all). It was also pretty unlikely that folks would have been comfortable with this anyway, since it would have been basically impossible to validate that this rule doesn't break any parsing code etc. So we won't be using this unfortunately :(

If this is blocking `0.49`, I'd be fine with reverting it or excluding it from the upcoming release. Alternatively we could work on improving some of the issues you pointed out. Let me know which approach you'd prefer.

**Comment by nicklockwood** [comment]
It's disabled by default anyway, so not a blocker, but I'd like to try to work towards a point where it could be enabled in future if possible.

**Pull request merged**